src/ru/animelib/src/eu/kanade/tachiyomi/animeextension/ru/animelib/Animelib.kt

```diff
@@ -0,0 +1,408 @@
+                }
+                Toast.makeText(screen.context, text, Toast.LENGTH_LONG).show()
+
+                preferences.edit().putBoolean(key, value).commit()
+            }
+        }.also(screen::addPreference)
+
+        if (!preferences.getBoolean(PrefConstants.PREF_USE_MAX_QUALITY_KEY, true)) {
+            MultiSelectListPreference(screen.context).apply {
+                key = PrefConstants.PREF_QUALITY_KEY
+                title = "Предпочитаемое качество"
+                entries = PrefConstants.PREF_QUALITY_ENTRIES
+                entryValues = PrefConstants.PREF_QUALITY_ENTRIES
+                summary = "При отсутствии нужного качества могут возникать ошибки!"
+                setDefaultValue(PrefConstants.PREF_QUALITY_ENTRIES.toSet())
+
+                setOnPreferenceChangeListener { _, newValue ->
+                    @Suppress("UNCHECKED_CAST")
+                    preferences.edit().putStringSet(key, newValue as Set<String>).commit()
+                }
+            }.also(screen::addPreference)
+        }
+
+        SwitchPreferenceCompat(screen.context).apply {
+            key = PrefConstants.PREF_IGNORE_SUBS_KEY
+            title = "Игнорировать субтитры"
+            summary = "Исключает из списка озвучек субтитры"
+            setDefaultValue(PrefConstants.PREF_IGNORE_SUBS_DEFAULT)
+
+            setOnPreferenceChangeListener { _, newValue ->
+                preferences.edit().putBoolean(key, newValue as Boolean).commit()
+            }
+        }.also(screen::addPreference)
+
+        EditTextPreference(screen.context).apply {
+            key = PrefConstants.PREF_DUB_TEAM_KEY
+            title = "Предпочитаемые студии озвучки"
+            summary = "Список студий или ключевых слов через запятую (экспериментальная функция)"
+            setDefaultValue("")
+
+            setOnPreferenceChangeListener { _, newValue ->
+                preferences.edit().putString(key, newValue as String).commit()
+            }
+        }.also(screen::addPreference)
+    }
+
+    // =============================== Details ===============================
+    override fun animeDetailsRequest(anime: SAnime): Request {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("anime")
+        url.addPathSegment(anime.url)
+        url.addQueryParameter("fields[]", "genres")
+        url.addQueryParameter("fields[]", "summary")
+        url.addQueryParameter("fields[]", "authors")
+        url.addQueryParameter("fields[]", "publisher")
+        url.addQueryParameter("fields[]", "otherNames")
+        url.addQueryParameter("fields[]", "anime_status_id")
+
+        return GET(url.build())
+    }
+
+    override fun getAnimeUrl(anime: SAnime) = "$baseUrl/${anime.url}"
+
+    override fun animeDetailsParse(response: Response): SAnime {
+        val animeInfo = json.decodeFromString<AnimeInfo>(response.body.string())
+
+        return animeInfo.data.toSAnime()
```

Comment:
```suggestion
        return response.parseAs<AnimeInfo>().data.toSAnime()
```
You can use `Response.parseAs<T>` instead of using decodeFromString on the body string to make it a bit more concise

Reply:
done